packages/shared/src/components/alert/PointedAlert.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { CSSProperties, ReactElement, ReactNode, useRef } from 'react';
+import classNames from 'classnames';
+import XIcon from '../icons/Close';
+import { Button } from '../buttons/Button';
+import {
+  PointedAlertMessage,
+  PointedAlertCopy,
+  PointedAlertWrapper,
+  PointedAlertContainer,
+} from './common';
+import Pointer, { PointerColor } from './Pointer';
+
+interface ClassName {
+  container?: string;
+  pointer?: string;
+  wrapper?: string;
+  message?: string;
+  label?: string;
+}
+
+type OffsetXY = [number?, number?];
+
+export enum AlertPlacement {
+  Top = 'top',
+  Right = 'right',
+  Bottom = 'bottom',
+  Left = 'left',
+}
+
+const messageBorder: Record<PointerColor, string> = {
+  [PointerColor.Cabbage]: 'border-theme-status-cabbage',
+  [PointerColor.Success]: 'border-theme-status-success',
+};
+
+const pointerClasses = {
+  [AlertPlacement.Top]: 'rotate-180',
+  [AlertPlacement.Right]: '-rotate-90',
+  [AlertPlacement.Left]: 'rotate-90',
+};
+
+const alertContainerClasses: Record<AlertPlacement, string> = {
+  top: '-translate-y-full top-0 flex-col-reverse',
```

Comment:
Would we not want to use `[AlertPlacement.Top]` here to generalize it?

Reply:
Just to make it less verbose, on the other variable, it did not use the `Record` type since not all placements required a value (bottom placement). Though we can probably convert the variable I am mentioning, it will only have an empty string literal for its value `''`, which is probably much safer since it will require the introduction of new items in the enum to force a value and not be forgotten.

Reply:
Yeah that sounds good, it will make it more uniform that way.
Resolving all the ones logged below in that case